### PR TITLE
tokyo-tyrant: added patch to fix unix socket issue

### PR DESCRIPTION
**Pull request by bow-fujita**
ttserver doesn't open unix socket even with command options `-host /path/to/socket -p 0`.
I added the patch introduced in [Tokyo Tyrant patch fix unix sockets on Mac OSX Snow Leopard](http://actsasflinn.com/post/482955247/tokyo-tyrant-patch-unix-socket-snow-leopard).
I confirmed it worked with tokyotyrant-1.1.41 on Yosemite, although this article was posted 6 years ago and the patch was applied to tokyotyrant-1.1.40.

### CODE REVIEW

File: Library/Formula/tokyo-tyrant.rb [hunk 1]
```diff
@@ -15,6 +15,11 @@ class TokyoTyrant < Formula
     patch :DATA
   end
 
+  patch :p0 do
+    url 'https://gist.githubusercontent.com/actsasflinn/348633/raw/32561ebf6487650ce86d0184f0ca35bb8110d04c/macosx_snow_leopard_socket_fix.patch'
+    sha1 '2b73fe52c37cd41b8e13112c0bc083bce8ae3c9e'
+  end
```
Comment (mikemcquaid) [inline Library/Formula/tokyo-tyrant.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (bow-fujita) [inline Library/Formula/tokyo-tyrant.rb]: I couldn't figure out how to submit a patch to the upstream developers. Do I need to create an issue in https://github.com/Homebrew/homebrew/issues ?

Comment (mikemcquaid) [inline Library/Formula/tokyo-tyrant.rb]: Looks like emailing info@fallabs.com.

Comment (bow-fujita) [inline Library/Formula/tokyo-tyrant.rb]: Emailed them and created [another patch on Gist](https://gist.github.com/bow-fujita/a338e35b95605ad63147) to explain the issue.

File: Library/Formula/tokyo-tyrant.rb [hunk 2]
```diff
@@ -15,6 +15,13 @@ class TokyoTyrant < Formula
     patch :DATA
   end
 
+  patch do
+    # Patch to fix UNIX socket issue
+    # https://gist.github.com/bow-fujita/a338e35b95605ad63147
+    url 'https://gist.githubusercontent.com/bow-fujita/a338e35b95605ad63147/raw/3bd1bb84ea58f7a71a485fe032299def3a288576/tokyotyrant-fix-unix-socket.patch'
+    sha1 'dd76282b0ea99104372c8da09e564f308d83ffdf'
```
Comment (mikemcquaid) [inline Library/Formula/tokyo-tyrant.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (bow-fujita) [inline Library/Formula/tokyo-tyrant.rb]: I submitted the patch to info@fallabs.com a few days ago but they haven't replied so far.
The explanation has already added to the formula file.

Comment (mikemcquaid) [inline Library/Formula/tokyo-tyrant.rb]: We'll need to wait for them before including it.

**Comment by DomT4** [comment]
Any update on this?

**Comment by bow-fujita** [comment]
No response from info@fallabs.com so far.

**Comment by bfontaine** [comment]
This PR needs to be rebased on top of `master`. Do we have any update from upstream?

**Comment by bfontaine** [comment]
Ping?

**Comment by bow-fujita** [comment]
I have no reply from upstream for a long time, but let me contact them again.
If they don't reply for a week, we should regard upstream as inactive.

**Comment by bfontaine** [comment]
Thanks!

**Comment by DomT4** [comment]
I think this project is mostly dead upstream, to be honest. I thought about patching the Lua situation but there seemed so little chance of upstream ever accepting it I removed the option from the formula instead.

Per upstream:
> BTW, do you know Kyoto Tycoon? Actually, it is more powerful and convenient server than Tokyo Tyrant. At this distance of time, Kyoto Tycoon surpasses Tokyo Tyrant in every aspects. I strongly recommend you to use Kyoto Tycoon.

**Comment by bow-fujita** [comment]
No reply from the upstream either. I can say they have already been dead.
I knew about KyotoTycoon, but I needed TokyoTyrant at that time for one of my projects.

**Comment by tdsmith** [comment]
```
tim@rocketman:homebrew-core (master)$ brew uses tokyo-tyrant
homebrew/php/php53-tokyotyrant     homebrew/php/php54-tokyotyrant     homebrew/php/php55-tokyotyrant     homebrew/php/php56-tokyotyrant
```

:skull:?

**Comment by mikemcquaid** [comment]
@tdsmith Yeh, go for it (but you may need to speak to the PHP tap).

**Comment by DomT4** [comment]
:+1: :skull:

**Comment by apjanke** [comment]
Boneyarding in Homebrew/homebrew-core#816 and Homebrew/homebrew-boneyard#112

**Pull request closed without merging**